tests/components/wled/test_switch.py

```diff
@@ -71,117 +72,83 @@
-    await hass.async_block_till_done()
-
-    state = hass.states.get("switch.wled_rgb_light_sync_receive")
-    assert state.state == STATE_OFF
-
-    await hass.services.async_call(
-        SWITCH_DOMAIN,
-        SERVICE_TURN_ON,
-        {ATTR_ENTITY_ID: "switch.wled_rgb_light_sync_receive"},
-        blocking=True,
-    )
-    await hass.async_block_till_done()
-
-    state = hass.states.get("switch.wled_rgb_light_sync_receive")
-    assert state.state == STATE_ON
+    with patch("wled.WLED.sync") as sync_mock:
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TURN_OFF,
+            {ATTR_ENTITY_ID: "switch.wled_rgb_light_sync_receive"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        sync_mock.assert_called_once_with(receive=False)
+
+    with patch("wled.WLED.sync") as sync_mock:
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "switch.wled_rgb_light_sync_receive"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        sync_mock.assert_called_once_with(receive=True)
 
 
 async def test_switch_error(
     hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
 ) -> None:
     """Test error handling of the WLED switches."""
     aioclient_mock.post("http://example.local:80/json/state", exc=aiohttp.ClientError)
-    await init_integration(hass, aioclient_mock)
-
-    await hass.services.async_call(
-        SWITCH_DOMAIN,
-        SERVICE_TURN_ON,
-        {ATTR_ENTITY_ID: "switch.wled_rgb_light_nightlight"},
-        blocking=True,
-    )
-    await hass.async_block_till_done()
-    state = hass.states.get("switch.wled_rgb_light_nightlight")
-    assert state.state == STATE_UNAVAILABLE
-
-    await hass.services.async_call(
-        SWITCH_DOMAIN,
-        SERVICE_TURN_ON,
-        {ATTR_ENTITY_ID: "switch.wled_rgb_light_sync_send"},
-        blocking=True,
-    )
-    await hass.async_block_till_done()
-    state = hass.states.get("switch.wled_rgb_light_sync_send")
-    assert state.state == STATE_UNAVAILABLE
-
-    await hass.services.async_call(
-        SWITCH_DOMAIN,
-        SERVICE_TURN_OFF,
-        {ATTR_ENTITY_ID: "switch.wled_rgb_light_sync_receive"},
-        blocking=True,
-    )
-    await hass.async_block_till_done()
-    state = hass.states.get("switch.wled_rgb_light_sync_receive")
-    assert state.state == STATE_UNAVAILABLE
+    entry = await init_integration(hass, aioclient_mock)
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+
+    with patch("homeassistant.components.wled.WLEDDataUpdateCoordinator.async_refresh"):
+        await hass.services.async_call(
+            SWITCH_DOMAIN,
+            SERVICE_TURN_ON,
+            {ATTR_ENTITY_ID: "switch.wled_rgb_light_nightlight"},
+            blocking=True,
+        )
+        await hass.async_block_till_done()
+        assert not coordinator.last_update_success
```

Comment:
Please change this too.

Reply:
Yeah, I should not have done that before my morning coffee... sorry about that.